tests/commonTemplates_test.go

```diff
@@ -75,6 +78,65 @@ var _ = Describe("Common templates", func() {
 			err := apiClient.Get(ctx, testTemplate.GetKey(), testTemplate.NewResource())
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		It("[test_id:5545]did not create duplicate templates", func() {
+			// TODO: the template path is relative pointing to the /data directory right now.
+			expectedTemplates, err := commonTemplates.ReadTemplates(filepath.Join("../"+commonTemplates.BundleDir, "common-templates-"+commonTemplates.Version+".yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(expectedTemplates).NotTo(HaveLen(0))
+
+			liveTemplates := &templatev1.TemplateList{}
+			err = apiClient.List(ctx, liveTemplates,
+				client.InNamespace(strategy.GetTemplatesNamespace()),
+				client.MatchingLabels{
+					"template.kubevirt.io/version": commonTemplates.Version,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, template := range expectedTemplates {
```

Comment:
I would prefer the test not read the templates bundle file, and work only with the deployed templates,.
This keeps this test consistent for an upgrade scenario as well, when templates from a previous bundle (that do not exist in the current bundle) are deployed (consistency should be kept in that scenario as well)

Reply:
I thought so too, bit wanted to replicate the test first.
It seemed to me, that the bundle was used to make sure every template in the bundle is present as well which is why I kept it.
I'm happy to change it though.

Reply:
@kwiesmueller I would also go with Omer's comment about maintaining consistency and listing the templates from already deployed ones.
But also have an separate test for checking each template in the bundle is present

Reply:
https://github.com/kubevirt/ssp-operator/blob/master/internal/operands/common-templates/reconcile_test.go#L79
Looks like we already have a test checking all templates are there.
So this should be done now.